src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +346,55 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
                     }
                 }
 
+                string authenticatemethod = "maskinporten";
+
+                if (!string.IsNullOrEmpty(Request.Headers["X-Altinn-EnterpriseUser-Authentication"]))
+                {
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    byte[] decodedCredentials = Convert.FromBase64String(enterpriseUserHeader);
+                    string decodedString = Encoding.UTF8.GetString(decodedCredentials);
+                    string[] decodedStringArray = decodedString.Split(":");
+                    string usernameFromRequest = decodedStringArray[0];
+                    string password = decodedStringArray[1];
+
+                    string bridgeApiEndpoint = _generalSettings.BridgeAuthnApiEndpoint + "enterpriseuser";
+                    EnterpriseUserAuthenticationService enterpriseUserAuthenticationService = new EnterpriseUserAuthenticationService();
+
+                    EnterpriseUserCredentials credentials = new EnterpriseUserCredentials { UserName = usernameFromRequest, Password = password, OrganizationNumber = orgNumber };
+                    ConfiguredTaskAwaitable<HttpResponseMessage> response = enterpriseUserAuthenticationService.GetResponseMessage(credentials, bridgeApiEndpoint);
+
+                    var responseInfo = response.GetAwaiter().GetResult();
+                    string content = await response.GetAwaiter().GetResult().Content.ReadAsStringAsync();
```

Comment:
This looks suspicious. Why does GetResponseMessage return a ConfiguredTaskAwaitable instead of just a Task? Why isn't the call to GetResponseMessage awaited? The responseInfo-task is instead awaited twice. Why can't this just be 

```
HttpResponseMessage response = await enterpriseUserAuthenticationService.GetResponseMessage(credentials, bridgeApiEndpoint);
string content = await response.Content.ReadAsStringAsync();
```

Reply:
Thank you, my bad. I have rewritten this part in latest commit.